packages/reaction-shipping/server/methods.js

```diff
@@ -5,37 +5,43 @@ Meteor.methods({
   addShippingMethod: function (insertDoc, currentDoc) {
     check(insertDoc, Object);
     check(currentDoc, String);
-    if (!Roles.userIsInRole(Meteor.userId(), ['admin', 'shipping'])) {
-      return false;
+    if (!ReactionCore.hasPermission("shipping")) {
+      throw new Meteor.Error(403, "Access Denied");
     }
+
     return ReactionCore.Collections.Shipping.update({
-      '_id': currentDoc
+      _id: currentDoc
     }, {
       $addToSet: {
-        'methods': insertDoc
+        methods: insertDoc
       }
     });
   },
 
-  /*
-   * Update Shipping methods for a provider
+  /**
+   * updateShippingMethods
+   * @summary update Shipping methods for a provider
+   * @param {String} providerId
+   * @param {String} methodId
+   * @param {Object} updateMethod - updated method itself
+   * @return update result
    */
-  updateShippingMethods: function (docId, currentDoc, updateDoc) {
-    check(docId, String);
-    check(currentDoc, Object);
-    check(updateDoc, Object);
-    if (!Roles.userIsInRole(Meteor.userId(), ['admin', 'shipping'])) {
-      return false;
+  updateShippingMethods: function (providerId, methodId, updateMethod) {
+    check(providerId, String);
+    check(methodId, String);
+    check(updateMethod, Object);
+    if (!ReactionCore.hasPermission("shipping")) {
+      throw new Meteor.Error(403, "Access Denied");
```

Comment:
owners always have permissions to all, but admin don't specifically (although they can assign themselves)